src/main/java/org/cryptomator/cryptofs/health/dirid/OrphanDir.java

```diff
@@ -108,45 +110,47 @@ public void fix(Path pathToVault, VaultConfig config, Masterkey masterkey, Crypt
 		Path cipherRecoveryDir = dataDir.resolve(recoveryDirHash.substring(0, 2)).resolve(recoveryDirHash.substring(2)).toAbsolutePath();
 		Files.createDirectories(cipherRecoveryDir);
 
+		return cipherRecoveryDir;
+	}
+
+	// visible for testing
+	CryptoPathMapper.CiphertextDirectory prepareStepParent(Path cipherRecoveryDir, FileNameCryptor cryptor, String clearStepParentDirName) throws IOException {
 		//create "step-parent" directory to move orphaned files to
 		String cipherStepParentDirName = convertClearToCiphertext(cryptor, clearStepParentDirName, Constants.RECOVERY_DIR_ID);
 		Path cipherStepParentDirFile = cipherRecoveryDir.resolve(cipherStepParentDirName + "/" + Constants.DIR_FILE_NAME);
 		final String stepParentUUID;
 		if (Files.exists(cipherStepParentDirFile, LinkOption.NOFOLLOW_LINKS)) {
 			stepParentUUID = Files.readString(cipherStepParentDirFile, StandardCharsets.UTF_8);
 		} else {
-			Files.createDirectory(cipherStepParentDirFile.getParent());
+			Files.createDirectories(cipherStepParentDirFile.getParent());
```

Comment:
I'd prefer `createDirectory`, since it gives us a `FileAlreadyExistException` for free. And since our stepparent has a unique name, an existing directory should not be possible (or indicates some error).

Reply:
Consider the following edge case:
1. a fix() is incomplete
2. the step-parent becomes orphaned

Due to the incomplete fix() the (actual) orphan still exists and with `createDirectory` every subsequent fix() attempt fails, because the stepparent name is deterministically generated. In contrast, when using `createDirectories` the fix can just continue where it left.

Reply:
My opinion is that in general inside the `CRYPTMATOR_RECOVERY` dir whe should be as lenient as possible wiht existing resources ar any errors. If a user stores normal data inside, its his/her fault.

Reply:
I see, I didn't had in mind that the stepparent name is essentially the original dir id hash. I thought it would be "more unique" but I do see the benefits of continuing an incomplete fix in the same dir.